autoload/go/list.vim

```diff
@@ -107,13 +107,22 @@ function! go#list#JumpToFirst(listtype) abort
   endif
 endfunction
 
-" Clean cleans the location list
+" Clean cleans and closes the location list 
 function! go#list#Clean(listtype) abort
   if a:listtype == "locationlist"
     lex []
   else
     cex []
   endif
+
+  let autoclose_window = get(g:, 'go_list_autoclose', 1)
+  if autoclose_window
+    if a:listtype == "locationlist"
+      lclose
+    else
+      cclose
+    endif
+  endif
```

Comment:
Instead of adding this block, calling `go#list#Window(a:listtype)` would reduce code duplication...

Reply:
Agree, but `Window()` is doing something else as well and it's get bloated. I'll add a `Close()` function and move out the logic there.